contracts/wrapper/NameWrapper.sol

```diff
@@ -363,7 +391,62 @@ contract NameWrapper is
     }
 
     /**
-     * @notice Sets fuses of a name that you own the parent of. Can also be called by the owner of a .eth name
+     * @notice Upgrades a .eth wrapped domain by calling the wrapETH2LD function of the upgradeContract
+     *     and burning the token of this contract
+     * @dev Can be called by the owner of the name in this contract
+     * @param label Label as a string of the .eth name to upgrade
+     * @param wrappedOwner The owner of the wrapped name
+     */
+
+    function upgradeETH2LD(
+        string calldata label,
+        address wrappedOwner,
+        address resolver
+    ) public {
+        bytes32 labelhash = keccak256(bytes(label));
+        bytes32 node = _makeNode(ETH_NODE, labelhash);
+        (uint32 fuses, uint64 expiry) = _prepareUpgrade(node);
+
+        upgradeContract.wrapETH2LD(
+            label,
+            wrappedOwner,
+            fuses,
+            expiry,
+            resolver
+        );
+    }
+
+    /**
+     * @notice Upgrades a non .eth domain of any kind. Could be a DNSSEC name vitalik.xyz or a subdomain
+     * @dev Can be called by the owner or an authorised caller
+     * Requires upgraded Namewrapper to permit old Namewrapper to call `setSubnodeRecord` for all names
+     * @param parentNode namehash of the parent name
+     * @param label Label as a string of the name to upgrade
+     * @param wrappedOwner Owner of the name in this contract
+     * @param resolver resolver contract for this name
+     */
+
+    function upgrade(
+        bytes32 parentNode,
+        string calldata label,
+        address wrappedOwner,
+        address resolver
+    ) public {
+        bytes32 labelhash = keccak256(bytes(label));
```

Comment:
Doesn't this function need to check that the parent name has been upgraded first?

Reply:
Decided we're not going to do this in favour of having checks in the new wrapper to reduce gas on upgrading. Added notes on upgrading in the README